x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authz/store/RoleReferenceTests.java

```diff
@@ -68,13 +73,28 @@ public void testApiKeyRoleReference() {
 
     public void testCrossClusterAccessRoleReference() {
         final var roleDescriptorsBytes = new CrossClusterAccessSubjectInfo.RoleDescriptorsBytes(new BytesArray(randomAlphaOfLength(50)));
-        final var crossClusterAccessRoleReference = new RoleReference.CrossClusterAccessRoleReference(roleDescriptorsBytes);
+        final var crossClusterAccessRoleReference = new RoleReference.CrossClusterAccessRoleReference("user", roleDescriptorsBytes);
 
         final RoleKey roleKey = crossClusterAccessRoleReference.id();
         assertThat(roleKey.getNames(), hasItem("cross_cluster_access:" + roleDescriptorsBytes.digest()));
         assertThat(roleKey.getSource(), equalTo("cross_cluster_access"));
     }
 
+    public void testFixedRoleReference() throws ExecutionException, InterruptedException {
+        final RoleDescriptor roleDescriptor = RoleDescriptorTests.randomRoleDescriptor();
+        final String source = "source";
+        final var fixedRoleReference = new RoleReference.FixedRoleReference(roleDescriptor, source);
+
+        final var future = new PlainActionFuture<RolesRetrievalResult>();
+        fixedRoleReference.resolve(mock(RoleReferenceResolver.class), future);
+        final var actualRetrievalResult = future.get();
+        assertThat(actualRetrievalResult.getRoleDescriptors(), equalTo(Set.of(roleDescriptor)));
+
+        final RoleKey roleKey = fixedRoleReference.id();
+        assertThat(roleKey.getNames(), hasItem(roleDescriptor.getName()));
```

Comment:
`hasItem` does not prevent `roleKey.getNames()` from having more items. So I suggest
```suggestion
        assertThat(roleKey.getNames(), containsInAnyOrder(roleDescriptor.getName()));
```

Alternatively, we can just compare `RoleKey` itself with
```
assertThat(roleKey, equalTo(new RoleKey(Set.of(roleDescriptor.getName()), source)));
```